File: leetcode/0485-max-consecutive-ones/solution.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        
        // best: longest streak of 1s seen so far
        int best = 0;
        
        // curr: length of the current running streak of 1s
        int curr = 0;
        
        for (int x : nums) {
            if (x==1) {
                
                // extend the current streak 
                curr ++;
                
                // update the best answer if needed
                if (curr>best) best = curr;
                
            } else {
                
                // 0 breaks the streak
                curr=0;
            }
        }
        return best;
    }
}

File: leetcode/0485-max-consecutive-ones/solution.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        
        max_count = 0 
        count = 0
        n = len(nums)

        for i in range(n):

            if nums[i] == 0:

                max_count = max(max_count,count)
                count = 0

            elif i == n-1:

                if nums[n-1] == 1:

                    count = count + 1

                max_count = max(max_count,count)

            else:

                count = count + 1

        return max_count

            
